### PR TITLE
implement team membership provisioning

### PR DESCRIPTION
**Pull request by jakcinmarina**
add team membership provisioning - user can be added as member or maintainer

### CODE REVIEW

File: pkg/connector/team.go [hunk 1]
```diff
@@ -247,6 +249,89 @@ func (o *teamResourceType) Grants(ctx context.Context, resource *v2.Resource, pT
...
+func (o *teamResourceType) Grant(ctx context.Context, principal *v2.Resource, entitlement *v2.Entitlement) (annotations.Annotations, error) {
+	l := ctxzap.Extract(ctx)
+
+	if principal.Id.ResourceType != resourceTypeUser.Id {
+		l.Warn(
+			"github-connectorv2: only users can be granted team membership",
+			zap.String("principal_type", principal.Id.ResourceType),
+			zap.String("principal_id", principal.Id.Resource),
+		)
+	}
```
Comment (martinalbert) [inline pkg/connector/team.go:261]: Missing return here.

File: pkg/connector/team.go [hunk 2]
```diff
@@ -247,6 +249,89 @@ func (o *teamResourceType) Grants(ctx context.Context, resource *v2.Resource, pT
...
+	entitlement := grant.Entitlement
+	principal := grant.Principal
+
+	if principal.Id.ResourceType != resourceTypeUser.Id {
+		l.Warn(
+			"github-connectorv2: only users can have team membership revoked",
+			zap.String("principal_type", principal.Id.ResourceType),
+			zap.String("principal_id", principal.Id.Resource),
+		)
+	}
```
Comment (martinalbert) [inline pkg/connector/team.go:305]: Same here.

**Comment by jakcinmarina** [comment]
fixed

**Review by martinalbert — APPROVED** [review]
LGTM

**Pull request merged**